storage/client/src/main/java/com/azure/storage/blob/BlobClient.java

```diff
@@ -244,17 +314,25 @@ public void download(OutputStream stream) throws IOException {
      * @param timeout
      *         An optional timeout value beyond which a {@link RuntimeException} will be raised.
      */
-    public void download(OutputStream stream, ReliableDownloadOptions options, BlobRange range,
-            BlobAccessConditions accessConditions, boolean rangeGetContentMD5, Duration timeout) throws IOException {
-        Flux<ByteBuffer> data = blobAsyncClient
-            .download(range, accessConditions, rangeGetContentMD5, options, null /*context*/);
-
-        data = timeout == null
-            ? data
-            : data.timeout(timeout); //TODO this isn't doing what we want
+    public VoidResponse download(OutputStream stream, ReliableDownloadOptions options, BlobRange range,
```

Comment:
IIRC we also want a download overload that returns an InputStream

Reply:
I thought we talked about this and agreed not to do it for this preview? But right now it should as simple as returning the BlobInputStream. Either way I'll leave this item for now due to time constraints and you guys can add if needed.